view/frontend/web/js/components/steps/BaseShippingStep.vue

```diff
@@ -124,219 +240,30 @@ export default {
-        }
-      });
-
-      return object;
-    },
-    changeSelection(event) {
-      /**
-       * Method onchange select (country/region)
-       * Returning country regions if exists
-       *
-      **/
-
-      const getForm       = event.srcElement.parentElement.parentElement,
-            countryId     = getForm.querySelector('#country_id'),
-            eventSelectId = event.srcElement.id,
-            inputRegion   = getForm.querySelector('#region'),
-            regionId      = getForm.querySelector('#region_id');
-
-      if (countryId == getForm.querySelector('#' + eventSelectId)) {
-        const eventOptionValue = event.srcElement.selectedOptions[0].value,
-              propertyRegions  = this.returnCountryRegions(this.regionList, eventOptionValue);
-
-        this.getShippingMethods(eventOptionValue);
-
-        inputRegion.value = '';
-
-        if (propertyRegions.length > 1) {
-          regionId.innerHTML = propertyRegions.join(' ');
-
-          this.isRegionIdHidden = false;
-        } else {
-          this.isRegionIdHidden = true;
-        }
-      } else if (regionId == getForm.querySelector('#' + eventSelectId)) {
-        const eventOptionCountryId = event.srcElement.selectedOptions[0].dataset.countryid,
-              eventOptionValue     = event.srcElement.selectedOptions[0].value;
-
-        if (!countryId.querySelector(`option[value="${eventOptionCountryId}"]`).selected) {
-          const propertyRegions = this.returnCountryRegions(this.regionList, eventOptionCountryId);
-
-          regionId.innerHTML = propertyRegions.join(' ');
-
-          regionId.querySelector(`option[value="${eventOptionValue}"]`).selected = true;
-          countryId.querySelector(`option[value="${eventOptionCountryId}"]`).selected = true;
-
-          this.isRegionIdHidden = false;
-        }
-      }
-    },
-    returnCountryRegions(regions, optionToCompare) {
-      /**
-       * Rendering country region list
-       * Return in array and passing to select
-       * Need to replace in future
-       *
-      **/
-
-      let newRegionList = [];
-
-      newRegionList.push(`
-        <option value="">
-          <?= /* @escapeNotVerified */ __('Please select a region, state or province.'); ?>
-        </option>
-      `);
-
-      regions.forEach(region => {
-        if (region.country_id === optionToCompare) {
-          newRegionList.push(`
-            <option value="${region.value}" data-countryid="${region.country_id}">
-              ${region.label}
-            </option>
-          `);
-        }
-      });
-
-      return newRegionList;
+    onCountryChange() {
+      this.regions = this.$store.getters.regionsByCountryId(this.countryId);
+      this.$store.dispatch('updateShippingMethods', this.countryId)
     },
-    returnError(element, cssClass, text) {
-      // Initial validation in future
+    onFormSubmit() {
+      // under this form data
+      console.log(this)
+      //TODO: save form data to store
```

Comment:
Do you have an idea how to do this? I was thinking about it and not pretty sure I will have right

Reply:
Sure, I have an idea :D